sql/wsrep_trans_observer.h

```diff
@@ -384,6 +385,11 @@ static inline
 int wsrep_after_statement(THD* thd)
 {
   DBUG_ENTER("wsrep_after_statement");
+  /* we need this otherwise we get stuck in the "SET DEBUG_SYNC" itself */
```

Comment:
It might be better to move this `DBUG_SYNC()` point in `wsrep_mysql_parse()`, just before call to `wsrep_after_statement()`. Seems that this method is called from many contexts, and I'm not sure if `thd->lex` validity can be guaranteed in all of them.